memory/src/main/java/org/opensearch/ml/memory/action/conversation/CreateInteractionRequest.java

```diff
@@ -92,14 +127,52 @@ public ActionRequestValidationException validate() {
      * @throws IOException if something goes wrong reading from request
      */
     public static CreateInteractionRequest fromRestRequest(RestRequest request) throws IOException {
-        Map<String, String> body = request.contentParser().mapStrings();
         String cid = request.param(ActionConstants.CONVERSATION_ID_FIELD);
-        String inp = body.get(ActionConstants.INPUT_FIELD);
-        String prmpt = body.get(ActionConstants.PROMPT_TEMPLATE_FIELD);
-        String rsp = body.get(ActionConstants.AI_RESPONSE_FIELD);
-        String ogn = body.get(ActionConstants.RESPONSE_ORIGIN_FIELD);
-        String addinf = body.get(ActionConstants.ADDITIONAL_INFO_FIELD);
-        return new CreateInteractionRequest(cid, inp, prmpt, rsp, ogn, addinf);
+        XContentParser parser = request.contentParser();
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+
+        String input = null;
+        String prompt = null;
+        String rep = null;
```

Comment:
minor: if the word is not so long, let's avoid using abbreviation, which make the code hard to read.